test/test_intensify/test_successive_halving.py

```diff
@@ -0,0 +1,347 @@
+        self.stats = Stats(scenario=self.scen)
+        self.stats.start_timing()
+
+        self.logger = logging.getLogger(self.__module__ + "." + self.__class__.__name__)
+
+    def test_init_1(self):
+        """
+            test parameter initializations for successive halving - instance as budget
+        """
+        intensifier = SuccessiveHalving(
+            tae_runner=None, stats=self.stats,
+            traj_logger=TrajLogger(output_dir=None, stats=self.stats),
+            rng=np.random.RandomState(12345), deterministic=False, run_obj_time=False,
+            instances=[1, 2, 3], n_seeds=2, initial_budget=None, max_budget=None, eta=2)
+
+        self.assertEqual(len(intensifier.instances), 6)  # since instance-seed pairs
+        self.assertEqual(intensifier.initial_budget, 1)
+        self.assertEqual(intensifier.max_budget, 6)
+        self.assertEqual(intensifier.num_initial_challengers, 4)  # 2 iterations
+        self.assertFalse(intensifier.cutoff_as_budget)
+
+    def test_init_2(self):
+        """
+            test parameter initialiations for successive halving - cutoff as budget
+        """
+        intensifier = SuccessiveHalving(
+            tae_runner=None, stats=self.stats,
+            traj_logger=TrajLogger(output_dir=None, stats=self.stats),
+            rng=np.random.RandomState(12345), deterministic=True, run_obj_time=False,
+            cutoff=30, instances=[1], initial_budget=1, max_budget=10, eta=2)
+
+        self.assertEqual(len(intensifier.instances), 1)  # since instance-seed pairs
+        self.assertEqual(intensifier.initial_budget, 1)
+        self.assertEqual(intensifier.max_budget, 10)
+        self.assertEqual(intensifier.num_initial_challengers, 8)  # 4 iterations
+        self.assertTrue(intensifier.cutoff_as_budget)
+
+    def test_init_3(self):
+        """
+            test parameter initialiations for successive halving - runtime cutoff as budget (no param provided)
+        """
+
+        with self.assertRaisesRegex(ValueError,
+                                    "requires parameters initial_budget and max_budget/cutoff for intensification!"):
+            SuccessiveHalving(
+                tae_runner=None, stats=self.stats,
+                traj_logger=TrajLogger(output_dir=None, stats=self.stats),
+                rng=np.random.RandomState(12345), deterministic=True, run_obj_time=False,
+                cutoff=10, instances=[1])
+
+    def test_init_4(self):
+        """
+            test parameter initialiations for successive halving - eta < 1
+        """
+
+        with self.assertRaisesRegex(ValueError, 'eta must be greater than 1'):
+            SuccessiveHalving(
+                tae_runner=None, stats=self.stats,
+                traj_logger=TrajLogger(output_dir=None, stats=self.stats),
+                rng=np.random.RandomState(12345), deterministic=True, run_obj_time=False,
+                cutoff=10, instances=[1], eta=0)
+
+    def test_init_5(self):
```

Comment:
What's the difference to the test above?

Reply:
Yes. I wanted to make it another test case for wrong initializations. I will also combine these similar cases into a single suite.